mlir/include/mlir/IR/ValueRange.h

```diff
@@ -112,6 +112,10 @@ class OperandRangeRange final
 
 /// This class provides a mutable adaptor for a range of operands. It allows for
 /// setting, inserting, and erasing operands from the given range.
+///
+/// (Note that when using AttrSizedOperandSegments, making changes to a
+/// MutableOperandRange invalidates other MutableOperandRange's in memory; you
+/// must get a new mutable adaptor for the other operand ranges).
```

Comment:
This is true even without using `AttrSizedOperandSegments` I believe.

Look at the members:
```
  /// The owning operation of this range.
  Operation *owner;
  /// The start index of the operand range within the owner operand list, and
  /// the length starting from `start`.
  unsigned start, length;
  ```

So any addition or removal of an operand for the `owner` Operation would invalidate the range.